ddtrace/tracer/log.go

```diff
@@ -0,0 +1,103 @@
+
+package tracer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"runtime"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+)
+
+const (
+	unknown = "unknown"
+)
+
+type startupInfo struct {
+	Date                  string            `json:"date"`                    // ISO 8601 date and time of start
+	OSName                string            `json:"os_name"`                 // Windows, Darwin, Debian, etc.
+	OSVersion             string            `json:"os_version"`              // Version of the OS
+	Version               string            `json:"version"`                 // Tracer version
+	Lang                  string            `json:"lang"`                    // "Go"
+	LangVersion           string            `json:"lang_version"`            // Go version, e.g. go1.13
+	Env                   string            `json:"env"`                     // Tracer env
+	Service               string            `json:"service"`                 // Tracer Service
+	AgentURL              string            `json:"agent_url"`               // The address of the agent
+	AgentError            string            `json:"agent_error"`             // Any error that occurred trying to connect to agent
+	Debug                 bool              `json:"debug"`                   // Whether debug mode is enabled
+	AnalyticsEnabled      bool              `json:"analytics_enabled"`       // True if there is a global analytics rate set
+	SampleRate            string            `json:"sample_rate"`             // The default sampling rate for the rules sampler
+	SamplingRules         []SamplingRule    `json:"sampling_rules"`          // Rules used by the rules sampler
+	SamplingRulesError    string            `json:"sampling_rules_error"`    // Any errors that occurred while parsing sampling rules
+	Tags                  map[string]string `json:"tags"`                    // Global tags
+	RuntimeMetricsEnabled bool              `json:"runtime_metrics_enabled"` // Whether or not runtime metrics are enabled
+	HealthMetricsEnabled  bool              `json:"health_metrics_enabled"`  // Whether or not health metrics are enabled
+	ApplicationVersion    string            `json:"dd_version"`              // Version of the user's application
+	Architecture          string            `json:"architecture"`            // Architecture of host machine
+	GlobalService         string            `json:"global_service"`          // Global service string. If not-nil should be same as Service. (#614)
+}
+
+func agentReachable(endpoint string) error {
+	req, err := http.NewRequest("POST", endpoint, nil)
+	if err != nil {
+		return fmt.Errorf("cannot create http request: %v", err)
+	}
+	_, err = defaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func logStartup(t *tracer) {
+	if !boolEnv("DD_TRACE_STARTUP_LOGS", true) {
```

Comment:
I wonder if it'd make more sense to have this check in `Start` and have `logStartup` always work otherwise. Just a curiosity, I'll leave it up to you to decide if it's better.

Reply:
We could do that. It feels like `logStartup` should log the start up, and shouldn't be called if we don't want to log the startup.

At the same time, I can't think of any practical effect moving it would have, and it would be nice to keep the configuration (`DD_TRACE_STARTUP_LOGS`) with the code it affects.

I think I'll leave it here. If we find a compelling reason to move it, we can do that later.